### PR TITLE
feat: Log slicing progress

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-301

Changes:
- Slicing progress is logged with exponentially increasing intervals.

image

### CODE REVIEW

File: internal/pkg/cli/config/config.go [hunk 1]
```diff
@@ -88,6 +88,14 @@ func Parse(args []string) (Config, error) {
 		return cfg, fmt.Errorf("cannot parse flags: %w", err)
 	}
 
+	// Define mapstructure hooks
+	hooks := viper.DecodeHook(
+		mapstructure.ComposeDecodeHookFunc(
+			mapstructure.TextUnmarshallerHookFunc(),
+			mapstructure.StringToTimeDurationHookFunc(),
```
Comment (michaljurecko) [inline internal/pkg/cli/config/config.go:95]: Enabled support for `time.Duration` type by `StringToTimeDurationHookFunc`.

File: internal/pkg/slicer/rowsreader/progress/progress.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+func (r *Logger) Close() error {
+	r.timer.Stop()
+	return nil
+}
+
+func (r *Logger) log() {
+	r.lock.Lock()
+	r.logger.Infof(`%s %05.2f%%`, r.message, float64(r.read*100)/float64(r.total))
+	r.lock.Unlock()
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/slicer/rowsreader/progress/progress.go:56]: In Go, a lock must be used on any value that is accessed from multiple goroutines. This is rule enforced by the data racer checker.

In this case, nothing bad would happen if there was no lock on the reading size, but the Go datarace checker doesn't understand the logic of the app, so it has to be there.

TLDR: Are you accessing the value from multiple goroutines? Use a Lock or RWLock. You don't have to think about it extra.

File: internal/pkg/cli/config/config.go (hunk 1 above)
Comment (enumag) [reply internal/pkg/cli/config/config.go:95]: I'll check the docs today and ask you on Monday. 🙂

Comment (michaljurecko) [reply internal/pkg/cli/config/config.go:95]: ok
